app/controllers/incomeSources/add/BusinessAddedObligationsController.scala

```diff
@@ -0,0 +1,167 @@
+          val quarterlyDatesByYear: (Seq[DatesModel], Seq[DatesModel]) = quarterlyDates.partition(x => dateService.getAccountingPeriodEndDate(x.inboundCorrespondenceTo) == dateService.getAccountingPeriodEndDate(quarterlyDates.head.inboundCorrespondenceTo))
+
+          val eopsDates: Seq[DatesModel] = dates.filter(x => x.periodKey == "EOPS")
+
+          val finalDeclarationDates: Seq[DatesModel] = getFinalDeclarationDates
+
+          val showPreviousTaxYears: Boolean = addedBusiness.tradingStartDate.get.isBefore(getStartOfCurrentTaxYear)
+
+
+          Future {
+            if (isAgent) Ok(obligationsView(businessName, ObligationsViewModel(quarterlyDatesByYear._1.sortBy(_.periodKey), quarterlyDatesByYear._2.sortBy(_.periodKey), eopsDates, finalDeclarationDates, dateService.getCurrentTaxYearEnd()),
+              controllers.incomeSources.add.routes.BusinessAddedObligationsController.agentSubmit(), agentBackUrl, isAgent = true, showPrevTaxYears = showPreviousTaxYears))
+            else Ok(obligationsView(businessName, ObligationsViewModel(quarterlyDatesByYear._1, quarterlyDatesByYear._2, eopsDates, finalDeclarationDates, dateService.getCurrentTaxYearEnd()),
+              controllers.incomeSources.add.routes.BusinessAddedObligationsController.submit(), backUrl, isAgent = false, showPrevTaxYears = showPreviousTaxYears))
+          }
+      }
+    }
+  }
+
+  private def getObligationDates(id: String)(implicit user: MtdItUser[_], ec: ExecutionContext): Seq[DatesModel] = {
+    Await.result(nextUpdatesService.getNextUpdates() map {
+      case NextUpdatesErrorModel(code, message) => Logger("application").error(
+        s"[BusinessAddedObligationsController][handleRequest] - Error: $message, code $code")
+        itvcErrorHandler.showInternalServerError()
+        Seq.empty
+      case NextUpdateModel(start, end, due, obligationType, dateReceived, periodKey) =>
+        Seq(DatesModel(start, end, due, obligationType, periodKey))
+      case ObligationsModel(obligations) =>
+        obligations.filter(x => x.identification == id).flatMap(obligation => obligation.obligations.map(x => DatesModel(x.start, x.end, x.due, x.obligationType, x.periodKey)))
+    }, Duration(100, MILLISECONDS))
+  }
+
+  private def getFinalDeclarationDates(implicit user: MtdItUser[_], ec: ExecutionContext): Seq[DatesModel] = {
+    Await.result(nextUpdatesService.getNextUpdates() map {
+      case model: ObligationsModel => model.allCrystallised map {
+        source => DatesModel(source.obligation.start, source.obligation.end, source.obligation.due, source.obligation.obligationType, source.obligation.periodKey)
+      }
+    }, Duration(100, MILLISECONDS))
+  }
+
+  private def getStartOfCurrentTaxYear: LocalDate = {
+    val currentDate = dateService.getCurrentDate()
+    if (currentDate.isBefore(LocalDate.of(currentDate.getYear, APRIL, 6))) LocalDate.of(currentDate.getYear - 1, APRIL, 6)
```

Comment:
manipulation with the dates, where required a new logic, should be moved under DateServiceInterface